akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonMigration.scala

```diff
@@ -22,11 +22,22 @@ import akka.util.unused
 abstract class JacksonMigration {
 
   /**
-   * Define current version. The first version, when no migration was used,
-   * is always 1.
+   * Define current version, that is, the value used when serializing new data. The first version, when no
+   * migration was used, is always 1.
    */
   def currentVersion: Int
 
+  /**
+   * Define the supported forward version this migration can read (must be greater or equal than `currentVersion`).
+   * If this value is different from [[currentVersion]] a [[JacksonMigration]] may be required to downcast
+   * the received blob to the current schema.
```

Comment:
json instead of blob?

Reply:
I didn't want to pick a format. Given this is Jackson-specific but format agnostic (CBOR, JSON). I'm not a fan of `blob` either.

Reply:
```suggestion
   * the received payload to the current schema.
```

What about using the term `payload`?

Reply:
The transform is operating on a json tree. payload works for me.